sw/device/lib/flash_ctrl.c

```diff
@@ -131,6 +131,10 @@ int flash_read(uint32_t addr, part_type_t part, uint32_t size, uint32_t *data) {
   return get_clr_err();
 }
 
+void flash_cfg_scramble_en(bool en) {
+  REG32(FLASH_CTRL_SCRAMBLE_EN(0)) = en ? 1 : 0;
```

Comment:
Consider using the modern style with `mmio_region_t`.

Also, the ternary is hard to parse. Use something like `en & 1` instead.

Reply:
There's no DIF for flash control, so I suggest leaving the software as-is in this patch, and we can revisit it once we have a DIF.

Reply:
> Also, the ternary is hard to parse. Use something like `en & 1` instead.

Not a crucial point, but I think this is definitely a "hardware vs software engineer" thing - I find that `en & 1` much harder to parse :) - but then I am accustomed to a world of multi-nested ternaries...

Reply:
@mcy, to @lenary 's point, i was mostly just trying to maintain the current style until a DIF is officially created.
I'll address your other comments though.